### PR TITLE
Bits needed for W3DDisplay

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/client/displaystring.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ class DisplayString : public MemoryPoolObject
     virtual void Set_Font(GameFont *font) { m_font = font; }
     virtual GameFont *Get_Font() { return m_font; }
     virtual void Set_Word_Wrap(int length) = 0;
-    virtual void Set_Word_Wrap_Centered(char unk) = 0;
+    virtual void Set_Word_Wrap_Centered(bool unk) = 0;
```
Comment (tomsons26) [inline src/game/client/displaystring.h]: name the arg then, state/warp_centered/on

Comment (jonwil) [reply src/game/client/displaystring.h]: fixed.

File: src/game/client/videoplayer/videobuffer.cpp [hunk 2]
```diff
@@ -53,3 +56,133 @@ RectClass VideoBuffer::Rect(float x1, float y1, float x2, float y2)
...
+    if (m_surface) {
+        return m_surface->Lock((int *)&m_pitch);
+    }
+
+    return ptr;
+}
+
+void W3DVideoBuffer::Unlock()
+{
+    if (m_surface) {
```
Comment (tomsons26) [inline src/game/client/videoplayer/videobuffer.cpp]: != nullptr

File: src/game/client/videoplayer/videobuffer.cpp [hunk 3]
```diff
@@ -53,3 +56,133 @@ RectClass VideoBuffer::Rect(float x1, float y1, float x2, float y2)
...
+{
+    void *ptr = nullptr;
+
+    if (m_surface) {
+        Unlock();
+    }
+
+    m_surface = m_texture->Get_Surface_Level(0);
+
+    if (m_surface) {
```
Comment (tomsons26) [inline src/game/client/videoplayer/videobuffer.cpp]: != nullptr

File: src/game/client/videoplayer/videobuffer.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/client/videoplayer/videobuffer.cpp]: fixed.

File: src/game/client/videoplayer/videobuffer.cpp (hunk 3 above)
Comment (jonwil) [reply src/game/client/videoplayer/videobuffer.cpp]: fixed.

File: src/game/client/videoplayer/videobuffer.cpp [hunk 4]
```diff
@@ -53,3 +56,133 @@ RectClass VideoBuffer::Rect(float x1, float y1, float x2, float y2)
...
+        Unlock();
+        return false;
+    }
+}
+
+void W3DVideoBuffer::Free()
+{
+    Unlock();
+
+    if (m_texture) {
```
Comment (tomsons26) [inline src/game/client/videoplayer/videobuffer.cpp]: != nullptr

Comment (jonwil) [reply src/game/client/videoplayer/videobuffer.cpp]: fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**